pkg/controller/infrastructure/infraflow/access/networking_access.go

```diff
@@ -197,7 +199,7 @@ func (a *networkingAccess) UpdateRouter(desired, current *Router) (modified bool
 func (a *networkingAccess) AddRouterInterfaceAndWait(ctx context.Context, routerID, subnetID string) error {
 	info, err := a.networking.AddRouterInterface(routerID, routers.AddInterfaceOpts{SubnetID: subnetID})
 	if err != nil {
-		return err
+		return util.DetermineError(err, helper.KnownCodes)
```

Comment:
that's not necessary since it's used at the top level in https://github.com/gardener/gardener-extension-provider-openstack/blob/65f305e201b0c356bec885212a5a7bac904e701c/pkg/controller/infrastructure/actuator_reconcile.go#L20